Doppler.ContactPolicies.Business.Logic.Test/UserFeaturesServiceTest.cs

```diff
@@ -0,0 +1,89 @@
+                });
+            UserFeaturesServiceSettings usersApiUrl = new UserFeaturesServiceSettings { UsersApiURL = "http://test.com/" };
+
+            var httpClient = new HttpClient(httpMessageHandlerStub.Object);
+            var clientFactoryMock = new Mock<IHttpClientFactory>(MockBehavior.Strict);
+            clientFactoryMock.Setup(cf => cf.CreateClient(It.IsAny<string>())).Returns(httpClient);
+
+            var loggerStub = new Mock<ILogger<UserFeaturesService>>();
+
+            var userFeaturesServiceSettingsStub = new Mock<IOptions<UserFeaturesServiceSettings>>();
+            userFeaturesServiceSettingsStub.Setup(x => x.Value).Returns(usersApiUrl);
+
+            var sut = new UserFeaturesService(clientFactoryMock.Object, userFeaturesServiceSettingsStub.Object, loggerStub.Object);
+
+            // Act
+            var result = await sut.HasContactPoliciesFeatureAsync(It.IsAny<string>());
+
+            // Assert
+            Assert.Equal(expectedFeatureContactPolicies, result);
+        }
+
+        [Theory]
+        [InlineData(HttpStatusCode.Forbidden, HttpStatusCode.Forbidden)]
+        [InlineData(HttpStatusCode.NotFound, HttpStatusCode.NotFound)]
+        public async Task HasContactPoliciesFeatureAsync_Should_ReturnCorrectStatusCode_When_ThrowException(HttpStatusCode mockStatusCode, HttpStatusCode expectedStatusCode)
+        {
+            // Arrange
+            var httpMessageHandlerStub = new Mock<HttpMessageHandler>();
+            httpMessageHandlerStub.Protected()
+                .Setup<Task<HttpResponseMessage>>("SendAsync", ItExpr.IsAny<HttpRequestMessage>(), ItExpr.IsAny<CancellationToken>())
+                .ReturnsAsync(new HttpResponseMessage
+                {
+                    StatusCode = mockStatusCode
+                })
+                .Verifiable();
+            UserFeaturesServiceSettings usersApiUrl = new UserFeaturesServiceSettings { UsersApiURL = "http://test.com/" };
+
+            var httpClient = new HttpClient(httpMessageHandlerStub.Object);
+            var clientFactoryMock = new Mock<IHttpClientFactory>(MockBehavior.Strict);
+            clientFactoryMock.Setup(cf => cf.CreateClient(It.IsAny<string>())).Returns(httpClient).Verifiable();
+
+            var loggerStub = new Mock<ILogger<UserFeaturesService>>();
+
+            var userFeaturesServiceSettingsStub = new Mock<IOptions<UserFeaturesServiceSettings>>();
+            userFeaturesServiceSettingsStub.Setup(x => x.Value).Returns(usersApiUrl);
+
+            var sut = new UserFeaturesService(clientFactoryMock.Object, userFeaturesServiceSettingsStub.Object, loggerStub.Object);
+
+            // Act
+            Action act = async () => await sut.HasContactPoliciesFeatureAsync(It.IsAny<string>());
+
+            // Assert
+            var ex = await Assert.ThrowsAsync<HttpRequestException>(async () => await sut.HasContactPoliciesFeatureAsync(It.IsAny<string>()));
+            Assert.Equal(expectedStatusCode, ex.StatusCode);
```

Comment:
Podríamos aprovechar para verificar el log ante una excepción.
```suggestion
            var accountName = "anyAccountName";

            // Act
            // Assert
            var ex = await Assert.ThrowsAsync<HttpRequestException>(async () => await sut.HasContactPoliciesFeatureAsync(accountName));
            loggerStub.Verify(
               x => x.Log(
                   It.Is<LogLevel>(l => l == LogLevel.Error),
                   It.IsAny<EventId>(),
                   It.Is<It.IsAnyType>((v, t) => v.ToString() == $"Failed to get contact policies feature with account name {accountName}"),
                   It.IsAny<Exception>(),
                   It.Is<Func<It.IsAnyType, Exception, string>>((v, t) => true)),
               Times.Once);
            Assert.Equal(expectedStatusCode, ex.StatusCode);
```